components/gantry/collectors.go

```diff
@@ -73,11 +71,22 @@ func newLengthsCollector(resource interface{}, params data.CollectorParams) (dat
 			}
 			return nil, data.FailedToReadErr(params.ComponentName, lengths.String(), err)
 		}
-		return Lengths{Lengths: v}, nil
+		return pb.GetLengthsResponse{
+			LengthsMm: scaleMetersToMm(v),
+		}, nil
 	})
 	return data.NewCollector(cFunc, params)
 }
 
+func scaleMetersToMm(meters []float64) []float64 {
+	ret := make([]float64, len(meters))
+	copy(ret, meters)
```

Comment:
Suuuper nitty, but this looks possible inefficient compared do a direct calculation? Since for each value are basically writing twice instead of once (if we skipped the copy and just did `ret[i] = meters[i] * 1000`)